cmd/gen-config_test.go

```diff
@@ -0,0 +1,187 @@
+
+type genConfig struct {
+	suite.Suite
+	assert *assert.Assertions
+}
+
+func (suite *genConfig) SetupTest() {
+	suite.assert = assert.New(suite.T())
+}
+
+func (suite *genConfig) cleanupTest() {
+	os.Remove("generatedConfig.yaml")
+}
+
+func (suite *genConfig) TestFileCacheConfigGen() {
+	defer suite.cleanupTest()
+
+	tempDir, _ := os.MkdirTemp("", "TestTempDir")
+	os.MkdirAll(tempDir, 0777)
+	defer os.RemoveAll(tempDir)
+
+	cmd := exec.Command("../blobfuse2", "gen-config", fmt.Sprintf("--component=%s", "file_cache"), fmt.Sprintf("--tmp-path=%s", tempDir))
+	_, err := cmd.Output()
+	suite.assert.Nil(err)
+
+	//Check if a file is generated named generatedConfig.yaml
+	suite.assert.FileExists("generatedConfig.yaml")
+
+	//check if the generated file is not empty
+	file, err := os.ReadFile("generatedConfig.yaml")
+	suite.assert.Nil(err)
+	suite.assert.NotEmpty(file)
+
+	//check if the generated file has the correct component
+	suite.assert.Contains(string(file), "file_cache")
+
+	//check if the generated file has the correct temp path
+	suite.assert.Contains(string(file), tempDir)
+}
+
+func (suite *genConfig) TestBlockCacheConfigGen() {
+	defer suite.cleanupTest()
+
+	tempDir, _ := os.MkdirTemp("", "TestTempDir")
+	os.MkdirAll(tempDir, 0777)
+	defer os.RemoveAll(tempDir)
+
+	cmd := exec.Command("../blobfuse2", "gen-config", fmt.Sprintf("--component=%s", "block_cache"), fmt.Sprintf("--tmp-path=%s", tempDir))
+	_, err := cmd.Output()
+	suite.assert.Nil(err)
+
+	//Check if a file is generated named generatedConfig.yaml
+	suite.assert.FileExists("generatedConfig.yaml")
+
+	//check if the generated file is not empty
+	file, err := os.ReadFile("generatedConfig.yaml")
+	suite.assert.Nil(err)
+	suite.assert.NotEmpty(file)
+
+	//check if the generated file has the correct component
+	suite.assert.Contains(string(file), "block_cache")
+
+	//check if the generated file has the correct temp path
+	suite.assert.Contains(string(file), tempDir)
+
+	cmd = exec.Command("../blobfuse2", "gen-config", fmt.Sprintf("--component=%s", "block_cache"))
+	_, err = cmd.Output()
+	suite.assert.Nil(err)
+
+	file, err = os.ReadFile("generatedConfig.yaml")
+	suite.assert.Nil(err)
+	//check if the generated file has no tmp path
+	suite.assert.NotContains(string(file), tempDir)
+}
+
+// test direct io flag
+func (suite *genConfig) TestDirectIOConfigGen() {
+	defer suite.cleanupTest()
+
+	cmd := exec.Command("../blobfuse2", "gen-config", fmt.Sprintf("--component=%s", "block_cache"), "--direct-io")
+
+	_, err := cmd.Output()
+	suite.assert.Nil(err)
+
+	//Check if a file is generated named generatedConfig.yaml
+	suite.assert.FileExists("generatedConfig.yaml")
+
+	//check if the generated file is not empty
+	file, err := os.ReadFile("generatedConfig.yaml")
+	suite.assert.Nil(err)
+	suite.assert.NotEmpty(file)
+
+	//check if the generated file has the correct direct io flag
+	suite.assert.Contains(string(file), "direct-io: true")
+}
+
+func (suite *genConfig) TestInvalidComponent() {
```

Comment:
if it is not file cache or block cache but certain custom component,
should we fail it or generate the remaining configurations?

Reply:
Right now we only support fc, bc if the user gives an unsupported component its better if we fail and inform them during config generation to avoid future complications.